Debian patches are maintained in Git: see Vcs-Git in debian/control.
--- dbus-sharp-0.7.0.orig/dbus-sharp-1.0.pc.in
+++ dbus-sharp-0.7.0/dbus-sharp-1.0.pc.in
@@ -8,4 +8,4 @@ Name: Managed DBus
 Description: Managed D-Bus IPC protocol library and CLR binding
 Version: @VERSION@
 URL: http://mono.github.com/dbus-sharp/
-Libs: -r:${libdir}/mono/@PACKAGE@-@API_VERSION@/dbus-sharp.dll
+Libs: -r:${libdir}/cli/@PACKAGE@-@API_VERSION@/dbus-sharp.dll
--- dbus-sharp-0.7.0.orig/src/dbus-sharp.dll.config
+++ dbus-sharp-0.7.0/src/dbus-sharp.dll.config
@@ -1,3 +1,14 @@
 <configuration>
   <dllmap dll="libsocket" os="!solaris" target="libc.so.6"/>
+  <dllmap dll="libc">
+    <dllentry os="solaris" dll="libsocket.so.1" name="socket" target="socket"/>
+    <dllentry os="solaris" dll="libsocket.so.1" name="connect" target="connect"/>
+    <dllentry os="solaris" dll="libsocket.so.1" name="bind" target="bind"/>
+    <dllentry os="solaris" dll="libsocket.so.1" name="listen" target="listen"/>
+    <dllentry os="solaris" dll="libsocket.so.1" name="accept" target="accept"/>
+    <dllentry os="solaris" dll="libsocket.so.1" name="getsockopt" target="getsockopt"/>
+    <dllentry os="solaris" dll="libsocket.so.1" name="setsockopt" target="setsockopt"/>
+    <dllentry os="solaris" dll="libsocket.so.1" name="recvmsg" target="recvmsg"/>
+    <dllentry os="solaris" dll="libsocket.so.1" name="sendmsg" target="sendmsg"/>
+  </dllmap>
 </configuration>
--- dbus-sharp-0.7.0.orig/src/Connection.cs
+++ dbus-sharp-0.7.0/src/Connection.cs
@@ -272,7 +272,7 @@ namespace DBus
 			rule.MessageType = MessageType.Signal;
 			rule.Fields.Add (FieldCode.Interface, new MatchTest (signal.Interface));
 			rule.Fields.Add (FieldCode.Member, new MatchTest (signal.Member));
-			rule.Fields.Add (FieldCode.Sender, new MatchTest (signal.Sender));
+			//rule.Fields.Add (FieldCode.Sender, new MatchTest (signal.Sender));
 			rule.Fields.Add (FieldCode.Path, new MatchTest (signal.Path));
 
 			Delegate dlg;
--- dbus-sharp-0.7.0.orig/src/BusObject.cs
+++ dbus-sharp-0.7.0/src/BusObject.cs
@@ -56,7 +56,10 @@ namespace DBus
 			rule.Fields.Add (FieldCode.Interface, new MatchTest (iface));
 			rule.Fields.Add (FieldCode.Member, new MatchTest (member));
 			rule.Fields.Add (FieldCode.Path, new MatchTest (object_path));
-			rule.Fields.Add (FieldCode.Sender, new MatchTest (alt_bus_name ?? bus_name));
+			// FIXME: Cause a regression compared to 0.6 as name wasn't matched before
+			// the problem arises because busname is not used by DBus daemon and
+			// instead it uses the canonical name of the sender (i.e. similar to ':1.13')
+			//rule.Fields.Add (FieldCode.Sender, new MatchTest (alt_bus_name ?? bus_name));
 
 			if (adding) {
 				if (conn.Handlers.ContainsKey (rule))
--- dbus-sharp-0.7.0.orig/src/Unix.cs
+++ dbus-sharp-0.7.0/src/Unix.cs
@@ -197,7 +197,7 @@ namespace DBus.Unix
 
 		// Solaris provides socket functionality in libsocket rather than libc.
 		// We use a dllmap in the .config to deal with this.
-		internal const string LIBSOCKET = "libsocket";
+		internal const string LIBSOCKET = "libc";
 
 		public const short AF_UNIX = 1;
 		// FIXME: SOCK_STREAM is 2 on Solaris
